### PR TITLE
More gossip cleanups

### PR DESCRIPTION
**Pull request by rustyrussell**
A weird valgrind fix, some file movements.

Gossipd now will refuse to even look at channel announcements until they're 6 deep.  As a side-effect, gossipd now knows when we're synced, and won't even ask for gossip before then (to avoid rejecting channels).

Finally, we no longer ask for a flood if the user doesn't support gossip_queries (but they all do anyway).

### CODE REVIEW

File: lightningd/channel_control.c [hunk 1]
```diff
@@ -376,9 +376,9 @@ void peer_start_channeld(struct channel *channel,
 
 	if (channel->scid) {
 		scid = *channel->scid;
-		/* Subtle: depth=1 at funding height. */
-		reached_announce_depth = get_block_height(ld->topology) + 1 >=
-				       short_channel_id_blocknum(&scid) + ANNOUNCE_MIN_DEPTH;
+		reached_announce_depth
+			= is_scid_depth_announcable(&scid,
```
Comment (darosior) [inline lightningd/channel_control.c]: Why not using directly the above pointer instead of a temporary variable ?

Comment (rustyrussell) [reply lightningd/channel_control.c]: Minimal change, really: the replaced code used &scid too.

**Comment by rustyrussell** [comment]
Rebase onto master...

File: bitcoin/short_channel_id.h [hunk 2]
```diff
@@ -47,6 +48,16 @@ static inline u16 short_channel_id_outnum(const struct short_channel_id *scid)
 	return scid->u64 & 0xFFFF;
 }
 
+/* Subtly, at block N, depth is 1, hence the -1 here. eg. 103x1x0 is announcable
+ * when height is 108. */
+static inline bool
+is_scid_depth_announcable(const struct short_channel_id *scid,
```
Comment (cdecker) [inline bitcoin/short_channel_id.h]: Did you mean: announceable :-)

File: common/gossip_constants.h [hunk 3]
```diff
@@ -29,15 +25,16 @@
  * The `message_flags` bitfield is used to indicate the presence of optional
  * fields in the `channel_update` message:
  * | Bit Position  | Name                      | Field                |
- * | ------------- | ------------------------- | -------------------- |
+ * | ------------- | ------------------------- | ----------------...- |
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply common/gossip_constants.h]: Actually, there are more `---` in the original. `...` is matched with "anything" when we check the specs.

**Comment by rustyrussell** [comment]
Trivial rebase, @cdecker spelling fix, and added a CHANGELOG.md line.

**Pull request merged**